tests/jest/plugins/plugin.text_selection.test.js

```diff
@@ -86,50 +153,76 @@ describe("Generic tests", () => {
     await br.textSelectionPlugin.createTextLayer({ $container, page: { index: 2, width: 100, height: 100 }});
-    expect($container.find(".textSelectionSVG").length).toBe(1);
-    expect($container.find(".BRparagElement").length).toBe(1);
-    expect($container.find(".BRwordElement").length).toBe(5);
-    expect($container.find(".BRwordElement").last()).not.toBe(" ");
+    expect($container.find(".BRtextLayer").length).toBe(1);
+    expect($container.find("p").length).toBe(1);
+    expect($container.find(".BRwordElement").length).toBe(3);
+    expect($container.find(".BRspace").length).toBe(2);
   });
 
-  test("createTextLayer creates an svg layer with paragraph with word with 5 params coordinates", async () => {
+  test("createTextLayer creates text layer with paragraph with word with 5 params coordinates", async () => {
     const $container = br.refs.$brContainer;
     sinon.stub(br.textSelectionPlugin, "getPageText")
       .returns($(new DOMParser().parseFromString(FAKE_XML_5COORDS, "text/xml")));
     await br.textSelectionPlugin.createTextLayer({ $container, page: { index: 3, width: 100, height: 100 }});
-    expect($container.find(".textSelectionSVG").length).toBe(1);
-    expect($container.find(".BRparagElement").length).toBe(1);
+    expect($container.find(".BRtextLayer").length).toBe(1);
+    expect($container.find("p").length).toBe(1);
     expect($container.find(".BRwordElement").length).toBe(1);
   });
 
+  test("createTextLayer handles multiple lines", async () => {
+    const $container = br.refs.$brContainer;
+    sinon.stub(br.textSelectionPlugin, "getPageText")
+      .returns($(new DOMParser().parseFromString(FAKE_XML_MULT_LINES, "text/xml")));
+    await br.textSelectionPlugin.createTextLayer({ $container, page: { index: 3, width: 100, height: 100 }});
+    expect($container.find(".BRtextLayer").length).toBe(1);
+    expect($container.find("p").length).toBe(1);
+    expect($container.find(".BRlineElement").length).toBe(3);
+    // Adds space at end of line; except last line/hyphens
+    expect($container.find("p").text()).toMatch(/another/);
+    expect($container.find("p").text()).toMatch(/Suppose /);
+    expect($container.find("p").text()).toMatch(/lastWord$/);
+    expect($container.find("p > br").length).toBe(1);
+  });
+
+  test("createTextLayer repairs trailing hyphens", async () => {
+    const $container = br.refs.$brContainer;
+    sinon.stub(br.textSelectionPlugin, "getPageText")
+      .returns($(new DOMParser().parseFromString(FAKE_XML_MULT_LINES, "text/xml")));
+    await br.textSelectionPlugin.createTextLayer({ $container, page: { index: 3, width: 100, height: 100 }});
+
+    expect($container.find(".BRwordElement--hyphen").length).toBe(1);
+    expect($container.find(".BRwordElement--hyphen").closest(".BRlineElement").text().endsWith(' ')).toBe(false);
+    expect($container.find(".BRwordElement--hyphen").closest(".BRlineElement").text().endsWith('-')).toBe(false);
```

Comment:
bc we draw w css right?

Reply:
Exactly! So it's not in the actual HTML text.